### PR TITLE
SD info: Add dynamic units and free %

### PR DESCRIPTION
**Pull request by non-bin**
# What's new

- Add dynamic storage units for free/total in settings->sd info
- Add free %

# Verification 

- Built and tested (64GB SD)

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/storage_settings/scenes/storage_settings_scene_sd_info.c [hunk 1]
```diff
@@ -24,13 +24,44 @@ void storage_settings_scene_sd_info_on_enter(void* context) {
             dialog_ex, "Try to reinsert\nor format SD\ncard.", 3, 19, AlignLeft, AlignTop);
         dialog_ex_set_center_button_text(dialog_ex, "Ok");
     } else {
+        char unit_kb[] = "KB";
```
Comment (skotopes) [inline applications/storage_settings/scenes/storage_settings_scene_sd_info.c:27]: No need to put it in ram, use const.

File: applications/storage_settings/scenes/storage_settings_scene_sd_info.c [hunk 2]
```diff
@@ -24,13 +24,44 @@ void storage_settings_scene_sd_info_on_enter(void* context) {
...
+        char unit_mb[] = "MB";
+        char unit_gb[] = "GB";
+
+        double sd_total_val = (double)sd_info.kb_total;
```
Comment (skotopes) [inline applications/storage_settings/scenes/storage_settings_scene_sd_info.c:31]: Using float point math here is kinda overkill.

File: applications/storage_settings/scenes/storage_settings_scene_sd_info.c [hunk 3]
```diff
@@ -24,13 +24,44 @@ void storage_settings_scene_sd_info_on_enter(void* context) {
...
+        char* sd_total_unit = unit_kb;
+        double sd_free_val = (double)sd_info.kb_free;
+        char* sd_free_unit = unit_kb;
+
+        if(sd_total_val > 1024) {
```
Comment (skotopes) [inline applications/storage_settings/scenes/storage_settings_scene_sd_info.c:36]: Better solution will be to implement formatting function that takes double* and const char**

File: applications/storage_settings/scenes/storage_settings_scene_sd_info.c (hunk 2 above)
Comment (skotopes) [reply applications/storage_settings/scenes/storage_settings_scene_sd_info.c:31]: Especially using double. Float should be more than enough, use cast at the last moment when passing it to string_printf.

**Review by skotopes — CHANGES_REQUESTED** [review]

**Comment by skotopes** [comment]
@non-bin do you plan to continue work on this PR?

**Comment by non-bin** [comment]
Yeah I will

**Comment by skotopes** [comment]
Feel free to reopen when it will be ready

**Pull request closed without merging**